moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -466,15 +465,31 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
 
 const TwistCommand Servo::toPlanningFrame(const TwistCommand& command)
 {
+  // Reverse the twist vector since we need it to be (angular, linear) for transforming using tf2.
+  Eigen::VectorXd reordered_twist(6);
+  reordered_twist.tail<3>() = command.velocities.head<3>();
+  reordered_twist.head<3>() = command.velocities.tail<3>();
```

Comment:
I wonder if it would work better if we did things like this: https://godbolt.org/z/oqs16ocfn

So basically:


```cpp
// Defined once
Eigen::VectorXf flip_indices{6};
flip_indices << 3, 4, 5, 0, 1, 2;

// In your loop
auto transformed_twist = command.velocities(flip_indices);
tf2::doTransform(transformed_twist, transformed_twist, body_to_space_tf);
return TwistCommand{ servo_params_.planning_frame, transformed_twist(flip_indices));
```

Reply:
I did not know this existed ! Super convenient. Thanks

Reply:
Gave this some more thought.
For both the current method and the method you suggested we end up creating 2 vector variables.
In the absence of an advantage, the current method seems more readable to me, since it is clear that we are moving the "angular" and "linear" parts.